tests/conftest.py

```diff
@@ -1,34 +1,71 @@
 """PyTest configuration module. Defines useful fixtures, command line args."""
+from __future__ import annotations
+
 import logging
+import tempfile
+from collections.abc import Generator
 from pathlib import Path
 
+import numpy as np
+import pandas as pd
 import pytest
+import sqlalchemy as sa
 
 logger = logging.getLogger(__name__)
 
 
-def pytest_addoption(parser: pytest.Parser) -> None:
-    """Add package-specific command line options to pytest.
-
-    This is slightly magical -- pytest has a hook that will run this function
-    automatically, adding any options defined here to the internal pytest options that
-    already exist.
-    """
-    parser.addoption(
-        "--sandbox",
-        action="store_true",
-        default=False,
-        help="Flag to indicate that the tests should use a sandbox.",
+@pytest.fixture(scope="session")
+def df1() -> pd.DataFrame:
+    """A dataframe with a named primary key."""
+    df = pd.DataFrame(
+        {
+            "a": np.random.rand(100).tolist(),
+            "b": np.random.randint(100, size=100).tolist(),
+            "c": np.random.choice(["a", "b", "c", "d"], size=100).tolist(),
+        }
     )
+    df.index.name = "pk"
+    return df
 
 
 @pytest.fixture(scope="session")
-def test_dir() -> Path:
-    """Return the path to the top-level directory containing the tests.
+def df2() -> pd.DataFrame:
+    """A dataframe with no primary key."""
+    return pd.DataFrame(
+        {
+            "d": np.random.rand(100).tolist(),
+            "e": np.random.randint(100, size=100).tolist(),
+            "f": np.random.choice(["a", "b", "c", "d"], size=100).tolist(),
+        }
+    )
 
-    This might be useful if there's test data stored under the tests directory that
-    you need to be able to access from elsewhere within the tests.
 
-    Mostly this is meant as an example of a fixture.
-    """
-    return Path(__file__).parent
+@pytest.fixture(scope="session")
+def temp_db(
+    df1: pd.DataFrame,
+    df2: pd.DataFrame,
+) -> Generator[tuple[str, str, str], None, None]:
+    """Create a temporary SQLite DB for use in testing."""
+    urlpath = Path(tempfile.mkstemp(suffix=".db")[1])
+    engine = sa.create_engine(f"sqlite:///{urlpath}")
+    con = engine.connect()
+    con.execute(
+        """CREATE TABLE temp (
+        pk BIGINT PRIMARY KEY,
+        a REAL NOT NULL,
+        b BIGINT NOT NULL,
+        c TEXT NOT NULL);"""
+    )
+    con.execute(
+        """CREATE TABLE temp2 (
+        d REAL NOT NULL,
+        e BIGINT NOT NULL,
+        f TEXT NOT NULL);"""
+    )
+    df1.to_sql("temp", con=engine, if_exists="append")
+    df2.to_sql("temp_nopk", con=engine, if_exists="append", index=False)
```

Comment:
I typically use a [context manager for sqlalchemy connections](https://docs.sqlalchemy.org/en/14/core/connections.html) because it automatically closes the connection.

Reply:
Ah yeah that would be better. I adapted this from the `intake-sql` tests. I'll change it.